docs/advanced-usage/compute-backends/slurm/index.md

```diff
@@ -6,30 +6,147 @@ backend Kubernetes and HTCondor.
 
 ## Authentication
 
-In order to use CERN Slurm cluster you need to be authenticated using
-Kerberos. [Generate keytab file](../../access-control/kerberos/index.md#generating-keytab-file)
+In order to use CERN Slurm cluster you need an access to Slurm Linux HPC resources.
+Please find more information about how to get access to the cluster in the official [CERN Batch Service documentation](https://batchdocs.web.cern.ch/linuxhpc/access.html).
+You also have to be authenticated using Kerberos. In order to do it please [generate keytab file](../../access-control/kerberos/index.md#generating-keytab-file)
 and [upload it and your CERN username as secrets to REANA](../../access-control/kerberos/index.md#uploading-secrets).
 
 ## Specifying compute backend
 
 In order to execute certain steps of a workflow on the CERN Slurm cluster
-you must specify ``slurmcern`` as the step's execution backend in the
-workflow specification.
-
-```yaml hl_lines="6"
-
-   # Serial example
-   ...
-   steps:
-      - name: reana_demo_helloworld_htcondorcern
-        environment: 'python:2.7-slim'
-        compute_backend: slurmcern
-        commands:
-            - python "${helloworld}"
+you must specify ``slurmcern`` as the step's ``compute_backend`` in the
+workflow specification. Please find [some examples](.#examples) how to do it in the bottom
+of the page.
+
+## Specifying environment
+
+There are several different ways to specify environments in the workflow
+specification for Slurm compute backend:
+
+- default way would be to specify a docker image which will be automatically converted to Singularity container
+- it is also possible to specify a Singularity image. Please note that the container
+file needs to be included into `reana.yaml` inputs and uploaded to the workspace before
+starting the workflow
+- you can also use Singularity from CVMFS unpacked images
+
+## Specifying Slurm parameters
+
+To set a specific partition or a maximum runtime limit for your job you can
+specify `slurm_partition` and `slurm_time` respectively in the specification
+of **each workflow step**. Please note that by default all the jobs will use
+`inf-short` partition and `60` seconds job time limit.
+
+Read more about the available partition values on [CERN Batch Docs](https://batchdocs.web.cern.ch/linuxhpc/resources.html).
+More information about runtime limit property format could be found on [official SLURM documentation](https://slurm.schedmd.com/sbatch.html).
+
+## Examples
+
```

Comment:
The following **CWL** workflow specification will dispatch the first "gendata" step of the [RooFit demo example](https://github.com/reanahub/reana-demo-root6-roofit) to the CERN Slurm compute backend, using a regular **Docker image** for the job environment:

```
...
```

The following **Serial** workflow specification will dispatch the first "gendata" step of the [RooFit demo example](https://github.com/reanahub/reana-demo-root6-roofit) to the CERN Slurm compute backend, using a **custom Singularity image** called `myimage_1_0.sif` from the workspace:

```
...
```

The following **Snakemake** workflow specification will dispatch the first "gendata" step of the [RooFit demo example](https://github.com/reanahub/reana-demo-root6-roofit) to the CERN Slurm compute backend, using a particular **Singularity image from CVMFS unpacked image area**:
```
...
```

The following **Yadage** workflow specification will dispatch the first "gendata" step of the [RooFit demo example](https://github.com/reanahub/reana-demo-root6-roofit) to the CERN Slurm compute backend, using a **custom Slurm partition** called `photon` and a **custom Slurm timeout** of five minutes: